doc/vim-go.txt

```diff
@@ -2490,7 +2490,15 @@ Show only variables on the right-hand side: >
   let g:go_debug_windows = {
         \ 'vars':  'rightbelow 60vnew',
   \ }
+<
+
+                                                *'g:go_debug_preserve_layout'*
 
+Preserve window layout in debugging mode. If explicitly set, it will override
+the preserving behavior when |'g:go_debug_windows'| is empty.
+>
```

Comment:
As worded, this sounds like the inverse of what it does. I believe it should indicate that it preserves window layout in debugging mode regardless of the value of g:go_debug_windows.

Also, there should be no need for a user to explicitly set it. Making the default value `0` should be sufficient and results in a user experience consistent with how vim-go works.

Reply:
I agree that the wording can be improved. However, a default of 0 would conflict with setting `g:go_debug_windows = {}` and break backwards compatibility.

The problem is this new setting controls the same window option (`:only`) as `g:go_debug_windows`, so there needs to be a way to determine precedence when their values indicate conflicting behavior. The default value of -1 (unset) makes this new setting backwards compatible because -1 means defer to `g:go_debug_windows` for `:only` behavior ie if `g:go_debug_windows` is empty, preserve layout. If not -1, the user explicitly set the value and it should take precedence.

This also allows it to work in conjunction with `g:go_debug_windows` - a user can set `g:go_debug_windows = {}` and `g:go_debug_preserve_layout = 0`, which would not display any debug windows and also not preserve their layout (I'm not sure why someone would want this, but the settings behavior would be consistent nonetheless).

g:go_debug_windows | g:go_debug_preserve_layout | only
---- | ---- | ----
{} | -1 | noonly
{} | 0 | only
{} | 1 | noonly
non-empty | -1 | only
non-empty | 0 | only
non-empty | 1 | noonly

Reply:
I think there was a typo in my original request. The logic should be `if !(empty(l:debugwindows) || l:debugpreservelayout)`. With that, there's no need to determine whether `g:go_debug_preserve_layout` was explicitly set.

When `go#config#DebugWindows()` returns an empty dict, then the current behavior on master is correct and should be preserved: `only` is not used. What you've asked for is a way to also skip `only` when `l:debugwindows` is not empty; the option you've introduced is useful for that purpose, but there's no need to determine whether the user explicitly set it; it only needs to be considered when `l:debugwindows` is not empty.

g:go_debug_windows | g:go_debug_preserve_layout | only
---- | ---- | ----
{} | 0 | noonly
{} | 1 | noonly
non-empty | 0 | only
non-empty | 1 | noonly

i.e. `g:go_debug_preserve_layout` should only be considered when `l:debugwindows` is not empty.